daemon/vpn/nordlynx/libtelio/libtelio.go

```diff
@@ -28,40 +28,59 @@ import (
 
 const (
 	// TelioLocalConfigName defines env key for local config value
-	TelioLocalConfigName = "TELIO_LOCAL_CFG"
+	TelioLocalConfigName     = "TELIO_LOCAL_CFG"
+	defaultHeartbeatInterval = 60 * 60
 )
 
 type state struct {
-	Nickname  string `json:"nickname"`
-	State     string `json:"state"`
-	PublicKey string `json:"public_key"`
-	IsVPN     bool   `json:"is_vpn"`
-	IsExit    bool   `json:"is_exit"`
+	Nickname  string
+	State     teliogo.NodeState
+	PublicKey string
+	IsVPN     bool
+	IsExit    bool
 }
 
-type event struct {
-	Body state `json:"body"`
-}
-
-type eventFn func(string)
-
 func maskPublicKey(event string) string {
 	expr := regexp.MustCompile(`"public_key":(\s)*"(.*?)"`)
 	return expr.ReplaceAllString(event, `"public_key":"***"`)
 }
 
-func eventCallback(states chan<- state) eventFn {
-	return func(s string) {
-		log.Println(internal.InfoPrefix + maskPublicKey(s))
```

Comment:
before every event received was logged, is it logged somewhere else or is not needed anymore?

Reply:
Nice catch. I fixed it. Now, I have to `json.Marshal` the event, so the field names are in PascalCase (there are no json tags on that structure) so I had to adjust the `maskPublicKey` as well